python/cuml/tests/test_device_selection.py

```diff
@@ -793,3 +812,40 @@ def test_nn_methods(train_device, infer_device):
     ref_output = ref_output.todense()
     output = output.todense()
     np.testing.assert_allclose(ref_output, output, rtol=0.15)
+
+
+@pytest.mark.parametrize('train_device', ['cpu', 'gpu'])
+@pytest.mark.parametrize('infer_device', ['cpu', 'gpu'])
+def test_hdbscan_methods(train_device, infer_device):
+
+    if train_device == "gpu" and infer_device == "cpu":
+        pytest.skip("Can't transfer attributes to cpu for now")
+
+    ref_model = refHDBSCAN(prediction_data=True,
+                           approx_min_span_tree=False,
+                           max_cluster_size=0,
+                           min_cluster_size=30)
+    ref_trained_labels = ref_model.fit_predict(X_train_blob)
+
+    from hdbscan.prediction import all_points_membership_vectors \
+        as cpu_all_points_membership_vectors, approximate_predict \
+        as cpu_approximate_predict
+    ref_membership = cpu_all_points_membership_vectors(ref_model)
+    ref_labels, ref_probs = cpu_approximate_predict(ref_model, X_test_blob)
```

Comment:
Should be good, but it would be nice to double check that no function ever modify the content of `X_train_blob` and `X_test_blob` used by other tests.

Reply:
Yes, it seems to be `const` since it's used by multiple functions with no issues